### DISCUSSION TITLE
Release v0.8.7 / 5.50.7 (pre-release)

### DISCUSSION THREAD

**Discussion by DavidXanatos** [sandboxie-plus/Sandboxie#981] [Announcements]
![0 5 5](https://user-images.githubusercontent.com/3890945/105014988-575cb680-5a41-11eb-90d5-d942d28184b2.png)

This build reworks some internal mechanisms and fixes a lot of bugs as well as some new features.

#### You can support the project through donations, any help will be greatly appreciated.

[<img src="https://www.paypalobjects.com/en_US/AT/i/btn/btn_donateCC_LG.gif">](https://xanasoft.com/go.php?to=donate)

# Changelog

### Added
- added option to always auto pick the DefautlBox sandbox [#959](https://github.com/sandboxie-plus/Sandboxie/issues/959)
-- when this option is enabled the normal behavioure with a box selection dialog can be achived holding down CTRL
- added option to hide boxed form the run in box dialog
-- usefull to avoidl listing for example insecure compatybility test boxes
- added box options to sys tray [#439](https://github.com/sandboxie-plus/Sandboxie/issues/439) [#272](https://github.com/sandboxie-plus/Sandboxie/issues/272)

### Changed
- changed default terminate all boxed processes key from Ctrl+Break to Ctrl+Alt+Break [#974](https://github.com/sandboxie-plus/Sandboxie/issues/974)
- start.exe does no longer link in all that unused MFC code reduced file size form over 2.5MB to below 250KB
- updated the main sandman and tray icon [#963](https://github.com/sandboxie-plus/Sandboxie/issues/963)
- improved the bov view tree style

### Fixed
- added additional delay and retryes to kmdutill.exe to mitigate issues when unloading the driver [#968](https://github.com/sandboxie-plus/Sandboxie/issues/968)
- fixed issue with sbiectrl not being properly started after setup [#969](https://github.com/sandboxie-plus/Sandboxie/issues/969)
- fixed issue with explore sandboxed shell option [#972](https://github.com/sandboxie-plus/Sandboxie/issues/972)
- fixed issue when running sandman elevated [#932](https://github.com/sandboxie-plus/Sandboxie/issues/932)
- fixed new box selection dialog showing disabled boxes
- fixed issue updating box active status

### Removed
- removed Online Armor support as this product is deprecated since 2016

### Added
- added LibreWolf template (by Dyras) [#929](https://github.com/sandboxie-plus/Sandboxie/pull/929)

### Fixed
- fixed performance bug introduced in 0.8.5

### Added
- added global hotkey to terminate all boxed processes (by default Ctrl+Break)
- the Run Sandboxed dialog can now be handled by the Sandman UI.
- added "AllowBoxedJobs=y" allowing boxed processes to use nested jobs on Windows 8 and later
-- note: this allows Chrome and other programs to use the job system for additional isolation
- added librewolf.exe to the list of Firefox derivatives [#927](https://github.com/sandboxie-plus/Sandboxie/issues/927)
- added run regedit sandboxed menu command
- added new support settings tab to Sandman UI for updates and stuff
- added code integrity verification to Sbie service and UI
- added template for Vivaldi Notes (by isaak654) [#948](https://github.com/sandboxie-plus/Sandboxie/issues/948)
- added LibreWolf template (by Dyras) [#929](https://github.com/sandboxie-plus/Sandboxie/pull/929)

### Changed
- Replaced the Process List used by the driver with a much faster Hash Map implementation
-- Note: this change provides an almost static system call speed of 1.2us irregardless of the running process count
-- The old list, with 100 programs running required: 4.5µs; with 200: 12µs; and with 300: 18µs per syscall
-- Note: some of the slowdown was affecting also non sandboxed applications due to how the driver handles certain callbacks
- Replaced the per-process Thread List used by the driver with a much faster Hash Map implementation
- Replaced configuration section list with a hash map to improve configuration performance, and increased line limit to 100000
-- not yet enabled in production build
- the presence of default box is only checked on connect
- the portable dir dialog now shows the directory [#924](https://github.com/sandboxie-plus/Sandboxie/issues/924)
- when terminated boxed processes now we first try doing that by terminating the job object
- the driver now by default can terminate problematic processes without the help of the service
- box delete routine now retries up to 10 times to fix [#954](https://github.com/sandboxie-plus/Sandboxie/issues/954)
- Replaced the Process List used by the service with a much faster Hash Map implementation
- Replaced the per-process Thread List used by the service with a much faster Hash Map implementation

### Fixed
- fixed faulty initialization in SetServiceStatus (by flamencist) [#921](https://github.com/sandboxie-plus/Sandboxie/issues/921)
- fixed buttons position in Classic UI settings (by isaak654) [#914](https://github.com/sandboxie-plus/Sandboxie/issues/914)
- fixed missing password length check in the Sandman UI [#925](https://github.com/sandboxie-plus/Sandboxie/issues/925)
- fixed issues opening job objects by name
- fixed missing permission check when reopening job object handles (thanks Diversenok)
- fixed issue with some Chromium 90+ hooks affecting PDF plugin in derived browsers [#930](https://github.com/sandboxie-plus/Sandboxie/issues/930) [#817](https://github.com/sandboxie-plus/Sandboxie/issues/817)
- fixed issues with reconnecting broken LPC ports used for communication with SbieSvc
- fixed minor setting issue [#957](https://github.com/sandboxie-plus/Sandboxie/issues/957)
- fixed minor UI issue with resource access COM settings [#958](https://github.com/sandboxie-plus/Sandboxie/issues/958)
- fixed an issue with NtQueryKey using NtQueryObject instead [#951](https://github.com/sandboxie-plus/Sandboxie/issues/951)
- fixed crash in key.c when failing to resolve key paths
- added workaround for topmost modality issue [#873](https://github.com/sandboxie-plus/Sandboxie/issues/873)
-- the notification window is not only topmost for 5 seconds
- fixed an issue deleting directories introduced in 5.49.5
- fixed an issue when creating box copies

### Removed
- removed switch for "BlockPassword=n" as it does not seem to be working [#938](https://github.com/sandboxie-plus/Sandboxie/issues/938)
-- it's recommended to use "OpenSamEndpoint=y" to allow for password change in windows 10

<hr /><em>This discussion was created from the release <a href='https://github.com/sandboxie-plus/Sandboxie/releases/tag/0.8.7'>Release v0.8.7 / 5.50.7</a>.</em>

**Comment by arie93** [comment]
Sandboxie Classic on this 5.50.7 version still breaks Internet Download Manager in Firefox. So, I'm still using v5.49.8 which still working fine on Internet Download Manager in Firefox.

**Reply by isaak654** [reply]
I can't find any report about Internet Download Manager not working in Firefox. Link?

**Reply by arie93** [reply]
Here's my report:
https://github.com/sandboxie-plus/Sandboxie/issues/991

**Reply by isaak654** [reply]
I replied you there with two different workarounds you can adopt, before an official fix.
